src/main/scala/it/pagopa/interop/purposeprocess/api/impl/PurposeApiServiceImpl.scala

```diff
@@ -135,35 +136,63 @@ final case class PurposeApiServiceImpl(
     val operationLabel = s"Updating Purpose $purposeId"
     logger.info(operationLabel)
 
+    def isDraft(purpose: PurposeManagementDependency.Purpose): Boolean = {
+      val states = purpose.versions.map(_.state)
+      states == Seq(PurposeManagementDependency.PurposeVersionState.DRAFT)
+    }
+
     val result: Future[Purpose] = for {
       organizationId <- getOrganizationIdFutureUUID(contexts)
       purposeUUID    <- purposeId.toFutureUUID
       purpose        <- purposeManagementService.getPurpose(purposeUUID)
       _              <- assertOrganizationIsAConsumer(organizationId, purpose.consumerId)
-      depPayload     <- PurposeUpdateContentConverter.apiToDependency(purposeUpdateContent).toFuture
+      depPayload     <-
+        if (isDraft(purpose))
+          PurposeUpdateContentConverter
+            .apiToDependency(purposeUpdateContent, RiskAnalysisValidation.validateOnlyFormal)
+            .toFuture
+        else
+          PurposeUpdateContentConverter.apiToDependency(purposeUpdateContent, RiskAnalysisValidation.validate).toFuture
       updatedPurpose <- purposeManagementService.updatePurpose(purposeUUID, depPayload)
-    } yield PurposeConverter.dependencyToApi(updatedPurpose)
+    } yield updatedPurpose.dependencyToApi(isRiskAnalysisValid = None)
 
     onComplete(result) { updatePurposeResponse[Purpose](operationLabel)(updatePurpose200) }
   }
 
-  override def getPurpose(id: String)(implicit
+  override def getPurpose(riskAnalysisFormCanPassValidation: Boolean, id: String)(implicit
     contexts: Seq[(String, String)],
     toEntityMarshallerPurpose: ToEntityMarshaller[Purpose],
     toEntityMarshallerProblem: ToEntityMarshaller[Problem]
   ): Route = authorize(ADMIN_ROLE, API_ROLE, SECURITY_ROLE, M2M_ROLE) {
     val operationLabel = s"Retrieving Purpose $id"
     logger.info(operationLabel)
 
+    def tryToValidateRiskAnalysisForm(
+      riskAnalysisForm: Option[PurposeManagementDependency.RiskAnalysisForm]
+    ): Future[Option[Boolean]] = {
```

Comment:
I would return a simple option here.
Wrapping in a Future is just for the for-compr aesthetic.